utils/zerovec/src/ule/mod.rs

```diff
@@ -214,11 +214,11 @@ where
 {
     #[inline]
     fn slice_to_unaligned(slice: &[Self]) -> Option<&[Self::ULE]> {
-        // This is safe because on little-endian platforms, the byte sequence of &[T]
-        // is equivalent to the byte sequence of &[T::ULE] by the contract of EqULE,
-        // and &[T::ULE] has equal or looser alignment than &[T].
-        let ule_slice =
-            unsafe { core::slice::from_raw_parts(slice.as_ptr() as *const Self::ULE, slice.len()) };
+        // char 4 bytes will be converted to 3 bytes of CharULE by skipping every 4th byte of information
+        let reduced_slice: &[u8] = slice.iter().copied().map(|c| c);
```

Comment:
Just a placeholder. Not sure what would be the most elegant mapper function here to skip every 4th byte.

Reply:
So where should the slice_to_unaligned() be moved to instead?

Reply:
It's not possible to implement `slice_to_unaligned` anymore, as the types have different sizes.

Reply:
Makes sense, so should I just remove the lines from the test at https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/src/ule/chars.rs#L127 ?

Reply:
You need to remove [this](https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/src/ule/chars.rs#L89-L91) impl (note the comment doesn't hold anymore), and the check you linked.

Reply:
(and revert this file)

Reply:
Yes, I realized after I commented 😬 did the changes.

Reply:
> You need to remove [this](https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/src/ule/chars.rs#L89-L91) impl (note the comment doesn't hold anymore), and the check you linked.

`ZeroVec::from_slice_or_alloc` seems to need this implementation? 🤔